include/alpaka/ctx/block/CtxBlockOacc.hpp

```diff
@@ -228,10 +228,17 @@ namespace alpaka
             //-----------------------------------------------------------------------------
             static auto declareVar(CtxBlockOacc<TDim, TIdx> const& smem) -> T&
             {
-                traits::SyncBlockThreads<CtxBlockOacc<TDim, TIdx>>::masterOpBlockThreads(smem, [&smem]() {
-                    smem.template alloc<T>();
-                });
-                return smem.template getLatestVar<T>();
+                auto* data = smem.template getVar<T>(TuniqueId);
+
+                if(!data)
+                {
+                    traits::SyncBlockThreads<CtxBlockOacc<TDim, TIdx>>::masterOpBlockThreads(smem, [&data, &smem]() {
+                        smem.template alloc<T>();
+                    });
+                    data = smem.template getVar<T>(TuniqueId);
```

Comment:
Here `getLatestVar` would be sufficient again. `SyncBlockThreads` acts like a `omp single`, except, that there is also a barrier in front, so the code is equivalent to the OpenMP5 version.

Reply:
Same as before.